### PR TITLE
feat(BUX-611): improve logging

### PR DESCRIPTION
**Pull request by jakubmkowalski**
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-js-client/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-js-client/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

Pino logger removed. 
Logger interface introduced.

### CODE REVIEW

File: src/authentication.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import bsv from 'bsv'
 import { RandomHex, Hash } from './utils'
 import { deriveHDPrivateChildKeyFromHex } from './utils/keys'
 import { signMessage } from './utils/sign'
-import logger from './logger'
+import { logger } from './logger'
```
Comment (arkadiuszos4chain) [inline src/authentication.ts:6]: I'm not sure, but with this approach users still can't configure/add a custom logger

Comment (chris-4chain) [reply src/authentication.ts:6]: I guess that the logger should be passed as an option/argument to the client and then used.

Something like that:

```typescript
interface Logger {
  debug(msg: string, ...args: any[]): void;
  info(msg: string, ...args: any[]): void;
  warn(msg: string, ...args: any[]): void;
  error(msg: string, ...args: any[]): void;
}

interface SimpleLoggerConfig {
  level: "debug" | "info" | "warn" | "error";
}

type LoggerConfig = Logger | SimpleLoggerConfig;

const isSimpleLoggerConfig = (config: LoggerConfig): config is SimpleLoggerConfig => {
  return (config as SimpleLoggerConfig).level !== undefined;
}

const nop = () => {}

export class SpvWalletClient {
  // (...)
  logger: Logger

  constructor(/* ... */ loggerConfig?: LoggerConfig) {
    // (...)

    if(loggerConfig == null) {
      this.logger = console
    }else if(isSimpleLoggerConfig(loggerConfig)) {
      const { level } = loggerConfig
      this.logger = {
        debug: level === 'debug' ? console.debug : nop,
        info: level === 'debug' || level === 'info' ? console.info : nop,
        warn: level === 'debug' || level === 'info' || level === 'warn' ? console.warn : nop,
        error: console.error,
      }
    } else {
      this.logger = loggerConfig
    }
    // (...)
}

**Review by chris-4chain — CHANGES_REQUESTED** [review]

File: src/client.ts [hunk 2]
```diff
@@ -785,7 +788,7 @@ export class SpvWalletClient {
   FinalizeTransaction(draftTransaction: DraftTransaction): string {
     if (!this.options?.xPriv) {
       const Err = new Error('cannot sign transaction without an xPriv')
-      logger.error(Err)
+      this.logger.error(Err.message)
```
Comment (chris-4chain) [inline src/client.ts:791]: For now, we don't use other methods than `.error()` but for future usage I think it will work perfectly 👍

**Review by chris-4chain — APPROVED** [review]

**Review by dorzepowski — APPROVED** [review]

**Comment by chris-4chain** [comment]
testing 🧪

**Pull request merged**